torch/_tensor_str.py

```diff
@@ -65,132 +65,128 @@ def set_printoptions(
         PRINT_OPTS.linewidth = linewidth
 
 
-def _get_min_log_scale():
-    min_positive = float_info.min * float_info.epsilon  # get smallest denormal
-    if min_positive == 0:  # use smallest normal if DAZ/FTZ is set
-        min_positive = float_info.min
-    return math.ceil(math.log(min_positive, 10))
+class _Formatter(object):
+    def __init__(self, tensor):
+        tensor = tensor.view(tensor.nelement())
```

Comment:
this isn't right, see the failing tests (you have to make the tensor contiguous).  What's the advantage of this over the old code given you may have to make the entire tensor contiguous?

Reply:
The old code makes a contiguous copy anyways. (And so will the new code) The main thing the new method is doing is making sure we're setting the item width to be exactly as large as we need. For example, we want `[1, 2]` and not `[ 1,  2]` if no elements are negative. Actually attempting to format all the numbers was the best way I could think of dealing with cases like negative 0.